lib/internal/Magento/Framework/App/Test/Unit/MaintenanceModeTest.php

```diff
@@ -58,7 +58,7 @@ public function testisOnWithIP()
         ];
         $this->flagDir->expects($this->exactly(2))->method('isExist')
             ->will(($this->returnValueMap($mapisExist)));
-        $this->assertFalse($this->model->isOn());
+        $this->assertTrue($this->model->isOn());
```

Comment:
Why need this change?

Reply:
This test was wrong, it basically stated:
If there is a maintenance flag file AND if there is an empty maintenance op file maintenance mode should be off.

This is not true. If the ip file is present but empty maintenance mode should be on, not off.

Reply:
Oh, nice catch 👍 Just still don't see corresponding changes in implementation which could invert this assertion.

Reply:
Wait... this cannot be right... Ha! https://github.com/magento/magento2/pull/10570/files#diff-4b45df6c223e8d98dcc949e45bfc4f1bR73 This is the wrong way around! Sorry.

Reply:
Just to clarify: this needs to be changed or currently logic is correct? How this test worked with incorrect previous logic though...

Reply:
Ping @ajpevers :)

Reply:
Hi @orlangur, sorry for the silence. I need to free some time to test why this worked without the PR and why it needs a change with the PR. I have it on my task list. Keep you posted.

Reply:
@orlangur I revisited this PR and it appears that this test is testing `!in_array("", [""])` which results in `false`.
So if your IP is `""` (the default value of the first and only `isON()` parameter) and the list of whitelisted IP's is `[""]` then you are marked as whitelisted and maintenance is not on for you. This is wrong.
If maintenance mode is on and the ip whitelist file is present but empty, maintenance mode should be on for everyone and not off. So this test is incorrect. I will fix this by adding an `array_filter` around the whitelisted IP's so that you can never compare against `[""]`, but in that case against `[]`, which will never have a false positive on `in_array` comparisons.